constellations/gemini-admin/src/semester/semesterEdit.js

```diff
@@ -45,23 +32,25 @@ export class SemesterEditPage extends React.Component {
                 });
             });
         }
-    }
+    };
 
-    handleChange(event, value) {
+    handleChange = (event, value) => {
         this.setState({ [value]: event.target.value });
-    }
+    };
 
-    handleIdChange(event, value) {
-        this.setState({ [value]: event.target.value });
+    handleIdChange = (event, value) => {
+        const newValue = event.target.value;
+        this.setState({ [value]: newValue });
         var possibleTitle =
-            event.target.value.substring(5) +
+            newValue.substring(5, 6).toUpperCase() +
```

Comment:
Add a comment here of what you're expecting. i.e. you're assuming the value looks like: `2020_fall` or 2021_winter.